docs/design/fabric-v2+/fpc-lifecycle-v2-new.puml

```diff
@@ -0,0 +1,139 @@
+/'
+    Copyright 2020 Intel Corporation
+    Copyright IBM Corp. All Rights Reserved.
+
+    SPDX-License-Identifier: Apache-2.0
+'/
+
+@startuml
+
+!pragma teoz true
+
+hide footbox
+title Foot Box removed
+
+title Fabric Private Chaincode - Lifecycle v2
+
+box "Org1"
+actor	Admin order 10
+entity "FPC_Peer1_Wrapper" as FPC_Peer1_Wrapper order 15
+participant Peer1 order 20
+entity "FPC_Peer2_Wrapper" as FPC_Peer2_Wrapper order 30
+participant Peer2 order 40
+end box
+
+box "Channel Abstraction"
+participant Channel order 70
+note over Channel
+    This box abstracts all other orgs and components.
+    Actions are supposed to be performed by channel participants.
+end note
+end box
+
+entity Orderer order 80
+
+note over Admin
+    Admin uses the fpc-peer wrappers, which abstract the Fabric Peer CLI.
+end note
+
+group chaincode package
+Admin -> FPC_Peer1_Wrapper : lifecycle chaincode package
+note right
+    **Chaincode package**
+    as tar.gz with
+    metadata.json = <path:(enclaves_path), type:(cc_lang), label:(cc_label), sgx_mode: (sim/hw)>
+    code.tar.gz : compressed file with enclave binary and mrenclave
+end note
+return
+end /' group chaincode package '/
+
+group install
+Admin -> FPC_Peer1_Wrapper ++: lifecycle chaincode install
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode install
+return packageId
+note over FPC_Peer1_Wrapper
+    Bookeeping operations to take into account that:
+    - FPC's initial version will run a chaincode on a designated peer
+    - the install lifecyle op is peer-specific
+    - the approveformyorg and commit ops are org-specific
```

Comment:
I think you are mixing up two things:
- we do bookkeeping as the cli doesn't have enough information from the `commit` parameters to figure out whether it is a FPC chaincode and which packages corresponds to it. That's why have to keep track of the mapping <code, package-id> in `install` and <package-id,id,version> in `approveformyorg`.

The other issue is that install is that not all action have to happen everywhere.  I think this is less an issue on bookkeeping but on requiring that approving and committing peers have the chaincode installed (which i'd argue is the normal expectation) and that we implicitly could define the committing peer as the designated-endorser (and hence piggy-back the corresponding enclave registration et al in commit)